lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -1358,16 +1358,16 @@ SwiftExpressionParser::ParseAndImport(
   invocation.getFrontendOptions().ModuleName = expr_name_buf;
   invocation.getIRGenOptions().ModuleName = expr_name_buf;
 
+  auto &lang_opts = invocation.getLangOptions();
   bool enable_bare_slash_regex_literals =
       m_sc.target_sp->GetSwiftEnableBareSlashRegex();
-  if (enable_bare_slash_regex_literals) {
-    invocation.getLangOptions().enableFeature(
-        swift::Feature::BareSlashRegexLiterals);
-  }
-  if (uint32_t version = m_expr.Language().version) {
-    invocation.getLangOptions().EffectiveLanguageVersion =
+  if (enable_bare_slash_regex_literals)
+    lang_opts.enableFeature(swift::Feature::BareSlashRegexLiterals);
+  if (uint32_t version = m_expr.Language().version)
+    lang_opts.EffectiveLanguageVersion =
         llvm::VersionTuple(version / 100, version % 100);
-  }
+  if (lang_opts.EffectiveLanguageVersion >= swift::version::Version({6}))
+    lang_opts.StrictConcurrencyLevel = swift::StrictConcurrency::Complete;
```

Comment:
does the compiler not do this when the version is set to `>=6`? If lldb is doing this, but the compiler isn't, is that going to cause some issues in some cases?

Reply:
also, I'm confused by the title. The diff appears to affect `StrictConcurrencyLevel`, and the language version already appears supported.

Reply:
Both excellent questions. I asked, and the compiler sets this flag in an ad-hoc fashion that isn't readily available to reuse. This also seems to be the only flag that behaves this way.

The title is wrong. This just adds a test.